core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -593,6 +596,14 @@ QservRestrictorPlugin::_newSecIndexRestrictor(
     query::QsRestrictor::Ptr p;
     std::shared_ptr<query::ColumnRef> secIndex =
         resolveAsColumnRef(context, cp->left);
+    if (secIndex)  {
+        std::string theDirTable =
+            context.cssFacade->getDirTable(secIndex->db, secIndex->table);
+        if ( ! theDirTable.empty() ) {
+            LOGF_DEBUG("Setting dirTable for sIndex to %1%" % theDirTable);
+            secIndex->table = theDirTable;
+        }
+    }
```

Comment:
I don't think this is the correct fix:
- You are setting the secondary index table without setting the secondary index db. Child tables do not have to be in the same database as their directors.
- You always set the secondary index table to the director table, if there is one. I think you should only do this if the column name matches the director column name for the child table.

This is also not the correct place for the fix - you should embed this in the other `_newSecIndexRestrictor` implementation (the one that handles `IN` clauses), as it is delegated to by this one. If you try to patch things up here, you will fail to fix the problem when clients use `IN` rather than `=`.

Finally, did you verify that `resolveAsColumnRef` returns a copy of the IR node from the query, and not a pointer to the actual node? If it does not, then changing the table / db could potentially hose subsequent analysis passes and we'll have to copy the IR node. If it does then we are fine (but please add a comment to the effect that we are mutating a copy).